plugins/chat/assets/javascripts/discourse/components/chat-channel-metadata.gjs

```diff
@@ -3,14 +3,25 @@ import { i18n } from "discourse-i18n";
 
 export default class ChatChannelMetadata extends Component {
   get lastMessageFormattedDate() {
-    return moment(this.args.channel.lastMessage.createdAt).calendar(null, {
+    const lastMessageDate = this.showThreadUnreadDate
+      ? this.args.channel.lastUnreadThreadDate
+      : this.args.channel.lastMessage.createdAt;
+
+    return moment(lastMessageDate).calendar(null, {
       sameDay: "LT",
       lastDay: `[${i18n("chat.dates.yesterday")}]`,
       lastWeek: "dddd",
       sameElse: "l",
     });
   }
 
+  get showThreadUnreadDate() {
+    return (
+      this.args.channel.lastUnreadThreadDate >
+      this.args.channel.lastMessage.createdAt
```

Comment:
Any chance of these being `undefined` at some point and causing errors? E.g. after just creating a channel?

Reply:
Good question. For channels with no threads it can be undefined, but it doesn't cause an error as it will just return false in those cases (ie. `undefined > new Date("Fri Nov 29 2024 12:00:00")`).

It's probably better to provide a placeholder date when there are no threads though, I think createdAt also does the same when joining a new channel so can probably return the same timestamp.